src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -0,0 +1,349 @@
+                                                                            Relationship_To_Predecessor__c,
+                                                                            Is_Deleted__c
+                                                                        FROM Data_Import_Object_Mapping__mdt
+                                                                        WHERE Data_Import_Object_Mapping_Set__c = :objectMappingSetId];
+        Map<String, Data_Import_Object_Mapping__mdt> diObjectNameToObjectMappingMap = new Map<String, Data_Import_Object_Mapping__mdt>();
+
+        for(Data_Import_Object_Mapping__mdt objectMapping : existingObjectMappings) {
+            diObjectNameToObjectMappingMap.put(objectMapping.DeveloperName, objectMapping);
+        }
+
+        return diObjectNameToObjectMappingMap;
+    }
+
+    /**
+     * @description Queries for default Field Mappings and builds a map of DeveloperName to Field Mapping record
+     * Used to check for existance of Field Mapping record without loop
+     * @param fieldMappingSetId Id of the default Field Mapping Set
+     * @return Collection of Field Mapping DeveloperName to Field Mapping record
+     */
+    private Map<String, Data_Import_Field_Mapping__mdt> getDefaultFieldMappings(Id fieldMappingSetId) {
+        List<Data_Import_Field_Mapping__mdt> existingFieldMappings = [SELECT Id,
+                                                                            Label,
+                                                                            DeveloperName,
+                                                                            Data_Import_Field_Mapping_Set__c,
+                                                                            Required__c,
+                                                                            Source_Field_API_Name__c,
+                                                                            Target_Field_API_Name__c,
+                                                                            Target_Object_Mapping__c,
+                                                                            Is_Deleted__c
+                                                                        FROM Data_Import_Field_Mapping__mdt
+                                                                        WHERE Data_Import_Field_Mapping_Set__c = :fieldMappingSetId];
+        Map<String, Data_Import_Field_Mapping__mdt> diFieldNameToFieldMappingMap = new Map<String, Data_Import_Field_Mapping__mdt>();
+
+        for(Data_Import_Field_Mapping__mdt fieldMapping : existingFieldMappings) {
+            diFieldNameToFieldMappingMap.put(fieldMapping.Source_Field_API_Name__c, fieldMapping);
+        }
+
+        return diFieldNameToFieldMappingMap;
+    }
+
+    /** @description Default Object Mapping Set record */
+    public static Data_Import_Object_Mapping_Set__mdt defaultObjectMappingSet {
+        get {
+            if(defaultObjectMappingSet == null) {
```

Comment:
I think our code standard demands a space between "if" and the paren - so "if (defaultObjectMappingSet == null) {"